### PR TITLE
Statically link with `SwiftPMDataModel` library

### LINKED ISSUE #274: carton 0.12.1 displays warnings
carton installed from brew runs into warnings or errors when issued. I see the version in the end. But there are other messages. Is this the normal behavior or is it an issue? I am not completely sure.

$ carton --version
objc[16139]: Class _TtC6Crypto18EllipticCurvePoint is implemented in both /usr/local/Cellar/carton/0.12.1/lib/libSwiftPMDataModel.dylib (0x1036f7ce0) and /usr/local/Cellar/carton/0.12.1/bin/carton (0x10102ff48). One of the two will be used. Which one is undefined.
objc[16139]: Class _TtC6Crypto27BoringSSLEllipticCurveGroup is implemented in both /usr/local/Cellar/carton/0.12.1/lib/libSwiftPMDataModel.dylib (0x1036f7da0) and /usr/local/Cellar/carton/0.12.1/bin/carton (0x101030008). One of the two will be used. Which one is undefined.
objc[16139]: Class _TtC6Crypto14ECDSASignature is implemented in both /usr/local/Cellar/carton/0.12.1/lib/libSwiftPMDataModel.dylib (0x1036f7e50) and /usr/local/Cellar/carton/0.12.1/bin/carton (0x1010300b8). One of the two will be used. Which one is undefined.
objc[16139]: Class _TtCV6Crypto25ArbitraryPrecisionInteger14BackingStorage is implemented in both /usr/local/Cellar/carton/0.12.1/lib/libSwiftPMDataModel.dylib (0x1036f7f50) and /usr/local/Cellar/carton/0.12.1/bin/carton (0x1010301b8). One of the two will be used. Which one is undefined.
objc[16139]: Class _TtC6Crypto28FiniteFieldArithmeticContext is implemented in both /usr/local/Cellar/carton/0.12.1/lib/libSwiftPMDataModel.dylib (0x1036f8018) and /usr/local/Cellar/carton/0.12.1/bin/carton (0x101030270). One of the two will be used. Which one is undefined.
**0.12.1**

### PR DESCRIPTION
**Pull request by yonihemi**
Resolves #274.

On macOS release binary grows from 27MB to 40MB.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
I guess this PR isn't available in 5.5 https://github.com/apple/swift-package-manager/pull/3520

**Comment by MaxDesiatov** [comment]
[5.6 is going to be branched real soon](https://forums.swift.org/t/swift-5-6-release-process/53412)

>On December 7, 2021 the release/5.6 branch will be cut in the `swift` repository and related project repositories.

**Comment by yonihemi** [comment]
Oh, nice one. Maybe you should also commit to 5.5?

**Comment by MaxDesiatov** [comment]
[I left a comment there](https://github.com/apple/swift-package-manager/pull/3520#issuecomment-853651028) asking whether SwiftPM release managers are interested in cherry-picking it for 5.5, but received no reply.

**Comment by MaxDesiatov** [comment]
Currently blocked by either https://github.com/apple/swift-package-manager/pull/3912 or https://github.com/apple/swift-package-manager/pull/3919

**Comment by MaxDesiatov** [comment]
Looks like there are enough breaking API changes in 5.6 to cause build errors. 😕

**Comment by MaxDesiatov** [comment]
Since the cherry-pick was merged into SwiftPM 5.5 branch, we can use that one and I can confirm that static link indeed fixes the issue.

**Review by MaxDesiatov — DISMISSED** [review]

**Comment by MaxDesiatov** [comment]
> On macOS release binary grows from 27MB to 40MB.

I'd assume this doesn't include `libSwiftPMDataModel.so` or a corresponding `.dylib` we had to bundle with the package. The total size of the previous executable binary with the library is very unlikely to be less than the statically linked version.

**Review by MaxDesiatov — APPROVED** [review]

**Pull request merged**
